docs/extending/extending_image_editor.md

```diff
@@ -0,0 +1,80 @@
+# Extending the Image Editor
+
+With the Image Editor, users can do basic image modifications.
+You can [modify the Image Editor's default appearance or behavior](../guide/image_editor.md).
+You can also extend it by adding custom features.
+For example, you can add a button that draws a dot at a random location on the image.
+Learn how to do this by following the steps below.
```

Comment:
As the scenario here is not really a real-life use case (I think?), how about emphasising that it's just an example?:
```suggestion

The following example shows how to extend the Image Editor
by adding a button that draws a dot at a random location on the image.
```